auction-app-backend/src/main/java/com/example/auctionapp/service/implementation/ProductServiceImpl.java

```diff
@@ -33,10 +28,31 @@ public ProductServiceImpl(final ProductRepository productRepository, final Categ
     }
 
     @Override
-    public Page<Product> getProducts(final int page, final int size) {
+    public Page<Product> getProducts(UUID categoryId, int page, int size) {
+        Pageable pageable = PageRequest.of(page, size);
+        Page<ProductEntity> productEntities;
+
+        if (categoryId != null) {
```

Comment:
Also this will be simplified with help of custom class for handling building filter parameters.